guides/tidymodels.qmd

```diff
@@ -0,0 +1,411 @@
+equivalent models. Here is the list:
+
+```{r}
+#| include: FALSE
+
+pf <- ls("package:parsnip")
+
+spark_engines <- map_dfr(pf, ~{
+  x <- get_from_env(.x)
+  if(!is.null(x)) x$model <- .x
+  x
+  }) %>% 
+  filter(engine == "spark") %>% 
+  select(-engine)
+
+supported_models <- spark_engines %>% 
+  mutate(supported = "Yes") %>% 
+  pivot_wider(
+    names_from = "mode", 
+    values_from = "supported", 
+    values_fill = ""
+    ) %>% 
+  mutate(title = get_title(model, "parsnip")) %>% 
+  select(title, model, everything()) %>% 
+  mutate(model = paste0("`", model, "()`"))
+```
+
+```{r}
+#| echo: FALSE
+supported_models %>% 
+  gt() %>% 
+  cols_label(
+    title = md("**Model**"),
+    model = md("**`parsnip` function**"),
+    classification = md("**Classification**"),
+    regression = md("**Regression**")
+  ) %>% 
+  fmt_markdown(model)
+```
+
+### Examples
+
+This article will use the same Spark session in all the examples.
+
+```{r}
+library(sparklyr)
+library(dplyr)
+
+sc <- spark_connect("local")
+```
+
+We will upload the `mtcars` data set to the Spark session:
+
+```{r}
+spark_mtcars <- copy_to(sc, mtcars)
+```
+
+A Linear Regression model is prepared against `spark_mtcars`:
+
+```{r}
+library(parsnip)
+
+mtcars_model <- linear_reg() %>%
+  set_engine("spark") %>%
+  fit(mpg ~ ., data = spark_mtcars)
+
+mtcars_model
+```
+
+It is also possible to see how `parsnip` plans to translate the model
+against the given engine. Use `translate()` so view the translation:
+
+```{r}
+linear_reg() %>%
+  set_engine("spark") %>%
+  translate()
+```
+
+Now, we will show an example with a classification model. We will fit a
+Random Forest model. To start, we will copy the `iris` data set to the
+Spark session:
+
+```{r}
+spark_iris <- copy_to(sc, iris)
+```
+
+We can prepare the model by piping the initial setup of 100 trees, then
+then to set the mode to "classification", and then the engine to "spark"
+and lastly, fit the model:
+
+```{r}
+iris_model <- rand_forest(trees = 100) %>% 
+  set_mode("classification") %>% 
+  set_engine("spark") %>% 
+  fit(Species ~., data = spark_iris)
+
+iris_model
+```
+
+## Model results with `broom`
+
+The `broom` package offers great ways to get summarized information
+about a fitted model. There is support for three `broom` functions in
+`sparklyr`:
+
+-   `tidy()` - Summarizes information about the components of a model. A
+    model component might be a single term in a regression, a single
+    hypothesis, a cluster, or a class.
+
+-   `glance()` - Returns a data frame with exactly one row of model
+    summaries. The summaries are typically goodness of fit measures,
+    p-values for hypothesis tests on residuals, or model convergence
+    information.
+
+-   `augment()` - It is similar to how `ml_predict()` works. Instead of
+    returning a vector of predictions, like `predict()` works,
+    `augment()` adds the prediction columns to the data set.
```

Comment:
```suggestion
-   `augment()` - Adds the prediction columns to the data set. This function 
     is similar to `ml_predict()`, but instead of returning only a vector of predictions 
     (like `predict()`), it adds the new column(s) to the data set.
    `augment()` 
```